wallet/src/lib.rs

```diff
@@ -776,14 +856,72 @@ struct GetEventsArgs {
 
 /// Return the recent events observed by this canister.
 #[query(guard = "is_custodian_or_controller")]
-fn get_events(args: Option<GetEventsArgs>) -> Vec<Event> {
+fn get_events128(args: Option<GetEventsArgs>) -> Vec<Event> {
     if let Some(GetEventsArgs { from, to }) = args {
         events::get_events(from, to)
     } else {
         events::get_events(None, None)
     }
 }
 
+#[query(guard = "is_custodian_or_controller")]
+fn get_events(args: Option<GetEventsArgs>) -> Vec<migrations::v1::V1Event> {
+    use migrations::v1::*;
+    let events = get_events128(args);
+    events
+        .into_iter()
+        .map(
+            |Event {
+                 id,
+                 timestamp,
+                 kind,
+             }| {
+                let kind = match kind {
+                    EventKind::AddressAdded { id, name, role } => {
+                        V1EventKind::AddressAdded { id, name, role }
+                    }
+                    EventKind::AddressRemoved { id } => V1EventKind::AddressRemoved { id },
+                    EventKind::CanisterCalled {
+                        canister,
+                        cycles,
+                        method_name,
+                    } => V1EventKind::CanisterCalled {
+                        canister,
+                        cycles: cycles.try_into().unwrap(),
```

Comment:
If we used .expect("Event cycles exceeded 64 bit limit.  Call get_events128") instead, would a caller get more information from the resulting panic?

Reply:
I had expected it to include what the conversion error actually was, but it doesn't, you're right.